test/over_react/component_declaration/builder_integration_tests/new_boilerplate/null_safety_validate_required_props_test.dart

```diff
@@ -0,0 +1,233 @@
+            ..requiredNonNullable = true
+            ..requiredNullable = null
+          )());
+        }, returnsNormally);
+      });
+
+      test('does not throw when the prop is set', () {
+        expect(() {
+          rtl.render((ComponentTest()
+            ..requiredNonNullable = true
+            ..requiredNullable = true
+          )());
+        }, returnsNormally);
+      });
+    });
+
+    test('@disableRequiredPropValidation annotation turns off validation for specific props', () {
+      expect(() {
+        rtl.render((ComponentTest()
+          ..requiredNonNullable = true
+          ..requiredNullable = true
+        )());
+      }, allOf(returnsNormally,
+        logsPropRequiredError('ComponentTestProps.disabledRequiredProp'),
+        logsPropRequiredError('ComponentTestProps.disabledNullableRequiredProp'),
+        logsPropRequiredError('ComponentTestProps.ref'),
+      ));
+    });
+
+    test('disableRequiredPropValidation method turns off validation for component usage', () {
+      expect(() {
+        rtl.render((ComponentTest()
+          ..disableRequiredPropValidation()
+        )());
+      }, allOf(returnsNormally,
+        logsPropRequiredError('ComponentTestProps.disabledRequiredProp'),
+        logsPropRequiredError('ComponentTestProps.disabledNullableRequiredProp'),
+        logsPropRequiredError('ComponentTestProps.ref'),
+        logsPropRequiredError('ComponentTestProps.requiredNonNullable'),
+        logsPropRequiredError('ComponentTestProps.requiredNullable'),
+      ));
+    });
+
+    group('required props in multiple mixins', () {
+      test('throw an error when a prop in the first mixin is missing', () {
+        expect(() {
+          rtl.render((MultipleMixinsTest()
+            ..requiredNullable = true
+            ..secondRequiredProp = true
+          )());
+        },
+            throwsA(isA<MissingRequiredPropsError>().having(
+                    (e) => e.toString(),
+                'toString value',
+                contains(
+                    'Required prop `requiredNonNullable` is missing.'))));
+      });
+
+      test('throw an error when a prop in the second mixin is missing', () {
+        expect(() {
+          rtl.render((MultipleMixinsTest()
+            ..requiredNullable = true
+            ..requiredNonNullable = true
+          )());
+        },
+            throwsA(isA<MissingRequiredPropsError>().having(
+                    (e) => e.toString(),
+                'toString value',
+                contains(
+                    'Required prop `secondRequiredProp` is missing.'))));
+      });
+
+      test('does not throw when all required props are set', () {
+        expect(() {
+          rtl.render((MultipleMixinsTest()
+            ..requiredNullable = true
+            ..requiredNonNullable = true
+            ..secondRequiredProp = true
+          )());
+        },
+          returnsNormally);
+      });
+    });
+  }, tags: 'ddc');
```

Comment:
Can we also verify that validation is disabled in dart2js? Doesn't have to be exhaustive will all the cases; just one test that missing required props don't cause failures should be sufficient.